addons/website/static/src/components/autocomplete_with_pages/url_autocomplete.js

```diff
@@ -0,0 +1,102 @@
+    static components = { AutoCompleteWithPages };
+
+    setup() {
+        super.setup();
+        this.rpc = useService("rpc");
+    }
+
+    _mapItemToSuggestion(item) {
+        return {
+            ...item,
+            classList: item.separator
+                ? "ui-autocomplete-category fw-bold text-capitalize p-2"
+                : "ui-autocomplete-item",
+        };
+    }
+
+    get dropdownClass() {
+        const classList = [];
+        for (const key in this.props.options?.classes) {
+            classList.push(key, this.props.options.classes[key]);
+        }
+        return classList.join(" ")
+    }
+
+    get sources() {
+        return [
+            {
+                optionTemplate: "website.AutoCompleteWithPagesItem",
+                options: async (term) => {
+                    if (term[0] === "#") {
+                        const anchors = await this.props.loadAnchors(
+                            term,
+                            this.props.options && this.props.options.body
+                        );
+                        return anchors.map((anchor) =>
+                            this._mapItemToSuggestion({ label: anchor, value: anchor })
+                        );
+                    } else if (term.startsWith("http") || term.length === 0) {
+                        // avoid useless call to /website/get_suggested_links
+                        return [];
+                    }
+                    if (this.props.options.isDestroyed?.()) {
+                        return [];
+                    }
+                    const res = await this.rpc("/website/get_suggested_links", {
+                        needle: term,
+                        limit: 15,
+                    });
+                    let choices = res.matching_pages;
+                    res.others.forEach((other) => {
+                        if (other.values.length) {
+                            choices = choices.concat(
+                                [{ separator: other.title, label: other.title }],
+                                other.values
+                            );
+                        }
+                    });
+                    return choices.map(this._mapItemToSuggestion);
+                },
+            },
+        ];
+    }
+
+    onSelect(selectedSubjection, { input }) {
+        if (!this.props.targetDropdown) {
+            this.props.onSelect(...arguments);
+            return;
+        }
+        const { value } = Object.getPrototypeOf(selectedSubjection);
+        input.value = value;
+        this.props.targetDropdown.value = value;
+        this.props.options.urlChosen?.();
+    }
+
+    onInput({ inputValue }) {
+        if (!this.props.targetDropdown) {
+            this.props.onInput(...arguments);
+            return;
+        }
+        this.props.targetDropdown.value = inputValue;
+        this.props.options.urlChosen?.();
+    }
+}
+
+UrlAutoComplete.props = {
```

Comment:
Randomly commenting here: we lost the "tab" capability to select an URL in the url dropdown, can we make it work?
If not, it's not the end of the world I guess

![image](https://github.com/odoo/odoo/assets/30048408/78553bb6-e8d8-4f46-b81e-e922ae1a0e8f)